src/Http/MiddlewareQueue.php

```diff
@@ -145,7 +145,7 @@ public function prepend($middleware)
      */
     public function insertAt($index, $middleware)
     {
-        array_splice($this->queue, $index, 0, $middleware);
+        array_splice($this->queue, $index, 0, [$middleware]);
```

Comment:
Nope, no check is necessary. `$middleware` will always represent a single middleware.